Prefer '=default' and implicit constructor fore more classes at Source/WebCore

vitorroriz opened the pull request:
#### b6966d585e085d03a2de9f7d1211c03741214a58
<pre>
Prefer &apos;=default&apos; and implicit constructor fore more classes at Source/WebCore
<a href="https://rdar.apple.com/129890943">rdar://129890943</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=275522">https://bugs.webkit.org/show_bug.cgi?id=275522</a>

Reviewed by Sammy Gill.

We prefer to let the compiler generate default constructors
rather than explicitly defining it. When the default
constructor is the only constructor we can omit it instead,
so it is implicitly generated.

* Source/WebCore/dom/NodeRareData.h:
(WebCore::NodeMutationObserverData::NodeMutationObserverData): Deleted.
* Source/WebCore/dom/SlotAssignment.h:
(WebCore::NamedSlotAssignment::Slot::Slot): Deleted.
* Source/WebCore/dom/TextEncoder.h:
(WebCore::TextEncoder::TextEncoder): Deleted.
* Source/WebCore/dom/TreeScopeOrderedMap.h:
* Source/WebCore/layout/formattingContexts/inline/AvailableLineWidthOverride.h:
* Source/WebCore/layout/integration/inline/InlineIteratorInlineBox.h:
(WebCore::InlineIterator::InlineBoxIterator::InlineBoxIterator): Deleted.
* Source/WebCore/layout/integration/inline/InlineIteratorTextBox.h:
(WebCore::InlineIterator::TextBoxIterator::TextBoxIterator): Deleted.
* Source/WebCore/rendering/AncestorSubgridIterator.cpp:
* Source/WebCore/rendering/TextPaintStyle.h:
(WebCore::TextPaintStyle::TextPaintStyle): Deleted.
* Source/WebCore/rendering/mathml/MathMLStyle.h:
* Source/WebCore/rendering/svg/legacy/SVGResourcesCycleSolver.h:
(WebCore::SVGResourcesCycleSolver::SVGResourcesCycleSolver): Deleted.
* Source/WebCore/style/StyleResolver.cpp:

Canonical link: <a href="https://commits.webkit.org/280226@main">https://commits.webkit.org/280226@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/59e5ce31cd1d3ef08a9aea5faa0ffb01146b7177

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/55730 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/35053 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/8197 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/58714 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/6161 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/57856 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/42675 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/6359 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/44879 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/4237 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/57759 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/32956 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/48039 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/26011 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/29743 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/5368 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/4304 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/51715 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/5635 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/60305 "Built successfully") | 
| | [⏳ 🛠 vision ](https://ews-build.webkit.org/#/builders/visionOS-1-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/5767 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/52306 "Passed tests") | 
| | [⏳ 🛠 vision-sim ](https://ews-build.webkit.org/#/builders/visionOS-1-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/48109 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/51799 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/12417 "Built successfully and passed tests") | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-1-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/30884 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/31969 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/33050 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/31716 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/dom/SlotAssignment.h:
```diff
@@ -107,7 +107,7 @@ class NamedSlotAssignment : public SlotAssignment {
         WTF_MAKE_FAST_ALLOCATED;
     public:
-        Slot() { }
+        Slot() = default;
```
Seems like we can just omit this line of code entirely instead.

darinadler commented on Source/WebCore/rendering/mathml/MathMLStyle.h:
```diff
@@ -34,7 +34,7 @@ namespace WebCore {
 class MathMLStyle: public RefCounted<MathMLStyle> {
 public:
-    MathMLStyle() { }
+    MathMLStyle() = default;
```
Seems like we can just omit this line of code entirely instead.

darinadler commented on Source/WebCore/dom/TextEncoder.h:
```diff
@@ -45,7 +45,6 @@ class TextEncoder : public RefCounted<TextEncoder> {
     RefPtr<Uint8Array> encode(String&&) const;
     EncodeIntoResult encodeInto(String&&, Ref<Uint8Array>&& destination);
 private:
-    TextEncoder() { };
```
Deleting this isn’t right, because then the `TextEncoding` default constructor will be public instead of private. So please add this back with `= default`.

vitorroriz replied:
ops, good catch, thanks!

darinadler commented on Source/WebCore/dom/SlotAssignment.h:
```diff
@@ -107,7 +107,6 @@ class NamedSlotAssignment : public SlotAssignment {
     struct Slot {
         WTF_MAKE_FAST_ALLOCATED;
     public:
-        Slot() { }
```
Please also delete the blank line after this.

darinadler commented on Source/WebCore/rendering/svg/legacy/SVGResourcesCycleSolver.h:
```diff
@@ -35,7 +35,6 @@ class SVGResourcesCycleSolver {
     static void resolveCycles(RenderElement&, SVGResources&);
 
 private:
-    SVGResourcesCycleSolver() { }
```
Deleting this isn’t right, because then the `SVGResourcesCycleSolver` default constructor will be public instead of private. So please add this back with `= default`.

vitorroriz committed:
Prefer '=default' and implicit constructor fore more classes at Source/WebCore
rdar://129890943
https://bugs.webkit.org/show_bug.cgi?id=275522

Reviewed by Sammy Gill.

We prefer to let the compiler generate default constructors
rather than explicitly defining it. When the default
constructor is the only constructor we can omit it instead,
so it is implicitly generated.

* Source/WebCore/dom/NodeRareData.h:
(WebCore::NodeMutationObserverData::NodeMutationObserverData): Deleted.
* Source/WebCore/dom/SlotAssignment.h:
(WebCore::NamedSlotAssignment::Slot::Slot): Deleted.
* Source/WebCore/dom/TextEncoder.h:
(WebCore::TextEncoder::TextEncoder): Deleted.
* Source/WebCore/dom/TreeScopeOrderedMap.h:
* Source/WebCore/layout/formattingContexts/inline/AvailableLineWidthOverride.h:
* Source/WebCore/layout/integration/inline/InlineIteratorInlineBox.h:
(WebCore::InlineIterator::InlineBoxIterator::InlineBoxIterator): Deleted.
* Source/WebCore/layout/integration/inline/InlineIteratorTextBox.h:
(WebCore::InlineIterator::TextBoxIterator::TextBoxIterator): Deleted.
* Source/WebCore/rendering/AncestorSubgridIterator.cpp:
* Source/WebCore/rendering/TextPaintStyle.h:
(WebCore::TextPaintStyle::TextPaintStyle): Deleted.
* Source/WebCore/rendering/mathml/MathMLStyle.h:
* Source/WebCore/rendering/svg/legacy/SVGResourcesCycleSolver.h:
(WebCore::SVGResourcesCycleSolver::SVGResourcesCycleSolver): Deleted.
* Source/WebCore/style/StyleResolver.cpp:

Canonical link: https://commits.webkit.org/280226@main
